File: Data Structures & Algorithms/sum-of-all-subset-xor-totals/submission-0.py
from typing import List

class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        result = 0  # This holds our running grand total

        def backtrack(index, current):
            nonlocal result  # 1. FIX: Allows us to modify the 'result' variable above
            
            if index == len(nums):
                result += current
                return

            # Choice 1: Include the element
            backtrack(index + 1, current ^ nums[index])
            # Choice 2: Exclude the element
            backtrack(index + 1, current)

        backtrack(0, 0)
        return result  # 2. FIX: Return 'result' instead of 'total'

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int[] res= new int[n];

        for(int i=0;i<n;i++){
            int pod=1;
            for(int j=0;j<n;j++){
                if(i!=j){
                    pod*=nums[j];
                }
            }
            res[i]=pod;
        } 
        return res;
    }
}  
